335 andreash94 customizable ai feedback

andreash94 opened the pull request:
Adding dropdown options for Tone, Voice, Level of detail when choosing an LLM to grade a student submission. These params get  passed into a prompt builder service that builds out the prompt. 

With the most recent merge there is an issue deploying locally with the import 'dart:html' as html; in the essay_edit_page.dart file. I commented that out and the corresponding calls to get this to work locally to see it run. 

Go to the essays page, and find a student submission. Under that you will see the dropdowns for choosing the LLM, tone, voice, level of detail. When you click grade those choices get sent in with the prompt.

andreash94 committed:
Adding prompt service to be able to pass customizable feedback arguments.

andreash94 committed:
Adding tone, voice, level of detail options for LLM grading

andreash94 committed:
restoring commented out code

ssung13 reviewed (commented):
Few things:

AI Selection:

I believe this was previous Cohort's code, but Perplexity is selected as default and it tries to run with Perplexity when "Grade" button is pressed, even though the option is grayed out.
It will be great to change it so that it defaults to whatever AI is available.

Grading inconsistency:

I'm not sure if this was intended, but if the Voice is set to Constructive, it tends to score less than what the supportive / neutral voice gives out. Probably adding something like

- Voice choice only affects phrasing style (Supportive = encouraging words, Neutral = factual wording, Constructive = improvement-focused) - it must not affect the score of the submission.

to the strict rules might help. Cody might have a better solution.

Other than these, I think it looks great :)

rappleb1 committed:
Merge branch 'team_e' into 335_andreash94_customizable_AI_Feedback

rappleb1 committed:
Merge branch 'team_e' into 335_andreash94_customizable_AI_Feedback

rappleb1 committed:
remove teamA

rappleb1 committed:
Merge branch 'team_e' into 335_andreash94_customizable_AI_Feedback

rappleb1 reviewed (changes requested):
What is the status of this feature and Google Classroom?

Codyjwhite reviewed (commented):
I took a look. I think we need better test data to see how much of an issue there is when it comes to the tone affecting the grade.

Some notes:
-The prompt for tone and voice could benefit from a more detailed description/examples of the output. As it stands the LLMs are gonna fill in the gaps as to what they think each word means which may result in inconsistent output across assignments and LLM Models.

-If voice and tone are affecting the grade, prompting 'may' fix it. If not my suggestion is to treat it as 2 separate prompts sent to the LLMs. The first grades the assignment objectively based on the rubric. Then a feedback prompt goes over it and provides feed back based on the objective grade.

rappleb1 committed:
Merge branch 'team_e' into 335_andreash94_customizable_AI_Feedback

andreash94 committed:
changing default LLM selection

andreash94 committed:
seperated grading from comment generation in the LLM prompting process to have objective grading

andreash94 commented:
Added default LLM to what is available. Changed up the LLM prompting to separate out the grading and the comment generation portions based on inputs.

andreash94 committed:
formatting fixes

rappleb1 commented on LearningLens2025/frontend/lib/services/prompt_builder_service.dart:
```diff
@@ -0,0 +1,63 @@
+
+**Part 2: Feedback**
+- After grading, generate feedback remarks according to the following options:
+  - Tone: $tone (Formal, Straightforward, Casual)
+  - Voice: $voice (Supportive, Neutral, Constructive)
```
I would revise the prompt for these just a bit. I am not seeing a whole lot of difference between the different tone and voice options. Like @Codyjwhite said I think providing a description of what each term means could help. Something similar to what you did for level of detail.

andreash94 committed:
updates to prompt

rappleb1 committed:
Merge remote-tracking branch 'origin/team_e' into 335_andreash94_customizable_AI_Feedback

rappleb1 reviewed (approved):
Seems improved from before, looks good for demo!
